src/openforms/utils/json_logic/descriptions.py

```diff
@@ -0,0 +1,325 @@
+    if not rest:
+        return gettext("{a} is less than or equal to {b}").format(a=a, b=b)
+    assert len(rest) == 1
+    return gettext(
+        "{a} is less than or equal to {b} and {b} is less than or equal to {c}"
+    ).format(a=a, b=b, c=rest[0])
+
+
+def op_not(operation: Operation, for_root=False) -> str:
+    arg = operation.arguments[0]
+    nested = _generate_description(arg)
+
+    if isinstance(arg, Primitive) or (
+        isinstance(arg, Operation) and arg.operator == "var"
+    ):
+        description = _("not {arg}").format(arg=nested)
+    else:
+        description = _("not ({arg})").format(arg=nested)
+    if not for_root:
+        description = f"({description})"
+    return description
+
+
+op_bool = FunctionLike(_("bool({args})"))
+op_modulo = JoinArgs(" % ")
+
+
+@add_boilerplate()
+def op_and(*args) -> str:
+    if not args:
+        return ""
+    description, *rest = args
+    for arg in rest:
+        description += _(" and {arg}").format(arg=arg)
+    return description
+
+
+@add_boilerplate()
+def op_or(*args) -> str:
+    if not args:
+        return ""
+    description, *rest = args
+    for arg in rest:
+        description += _(" or {arg}").format(arg=arg)
+    return description
+
+
+@add_boilerplate()
+def op_if(*nested) -> str:
+    num_args = len(nested)
+
+    if num_args == 0:
+        return "null"
+
+    elif num_args == 1:
+        description = nested[0]
+    elif num_args == 2:
+        condition, val = nested
+        description = _("{val} if {condition}, else null").format(
+            val=val, condition=condition
+        )
+    elif num_args == 3:
+        condition, val, else_val = nested
+        description = _("{val} if {condition}, else {else_val}").format(
+            val=val, condition=condition, else_val=else_val
+        )
+    else:  # pair them up into if/elif/elif/.../else
+        bits = []
+        for i in range(0, num_args - 1, 2):
+            condition, outcome = nested[i : i + 2]
+            bit = (
+                _("if {condition} then {outcome}")
+                if i == 0
+                else _("else if {condition} then {outcome}")
+            )
+            bits.append(bit.format(condition=condition, outcome=outcome))
+
+        if num_args % 2 == 1:
+            fallback = _("else {outcome}").format(outcome=nested[-1])
+            bits.append(fallback)
+
+        description = ", ".join(bits)
+
+    return description
+
+
+op_in = TemplatedArgs(_("{needle} in {haystack}"), arg_names=["needle", "haystack"])
+op_cat = FunctionLike(_("concatenate({args})"))
+op_sum = JoinArgs(" + ")
+op_multiply = JoinArgs(" * ")
+
+
+@add_boilerplate()
+def op_subtraction(*args) -> str:
+    if len(args) == 1:
+        return f"-{args[0]}"
+    return " - ".join(args)
+
+
+op_division = JoinArgs(" / ")
+op_min = FunctionLike(_("minimum of ({args})"))
+op_max = FunctionLike(_("maximum of ({args})"))
+
+
+def op_merge(operation: Operation, for_root=False) -> str:
```

Comment:
Also here, is `for_root` needed?

Reply:
Similar here, it's because of the Python callback protocol: https://github.com/open-formulieren/open-forms/blob/818c34b0e5a038c8f7c5457a0e8c4a9e298acd7c/src/openforms/utils/json_logic/descriptions.py#L18